ulixee/unblocked
# Feat: recursive dom profiler

Main focus of this PR is improving profiler and default-browser-emulator:
- Profiler now recursively goes into objects and not just prototypes alone and stores results in `_$otherInvocation` keys
- This is needed for things where `this` object changes output of of function: eg `Bluetooth.getAvailability` (illegal invocation) vs `navigator.bluetooth.getAvailability` (false or true)
-`removeUnsupportedPropertiesFromPolyfill` to remove things that emulator currently doesn't support -> these should be added in the future (failed promise results, functions that require arguments)
- Update instanceChangeExtractor with all new keys now in profiler data
- Look/store if function result is promise/async and mimic this in emulator
- Emulator creates single proxy for all things that needs proxies and depending on `this` changes output

Test fixes:
- Fix flacky mitm tests because of loading of favicon
- Navigation test 
  - Retry 3 times when not using mitm -> mitm disabled is currently flacky and might break test
  - popupPagePromise before loading so we don't miss event when registering callback too late

Other things done:
- MacOS use version 13 for github runners (should have better hardware)
- Make chrome 124 the default
- Update some docs
- Fixed chrome regex in bridgeUtils that ignore chrome >= 120
- Env var to run browserStackLocal and configure accordingly 
- Fix navigator plugin wrong call instead of apply invocation
